src/PhpSpec/ServiceContainer.php

```diff
@@ -1,128 +1,51 @@
 <?php
 
-/*
- * This file is part of PhpSpec, A php toolset to drive emergent
- * design by specification.
- *
- * (c) Marcello Duarte <marcello.duarte@gmail.com>
- * (c) Konstantin Kudryashov <ever.zet@gmail.com>
- *
- * For the full copyright and license information, please view the LICENSE
- * file that was distributed with this source code.
- */
-
 namespace PhpSpec;
 
-use InvalidArgumentException;
-
 /**
  * The Service Container is a lightweight container based on Pimple to handle
  * object creation of PhpSpec services.
  */
-class ServiceContainer
+interface ServiceContainer
 {
     /**
-     * @var array
-     */
-    private $parameters = array();
-
-    /**
-     * @var array
-     */
-    private $services = array();
-
-    /**
-     * @var array
-     */
-    private $prefixed = array();
-
-    /**
-     * @var array
-     */
-    private $configurators = array();
-
-    /**
      * Sets a param in the container
      *
      * @param string $id
-     * @param mixed  $value
+     * @param mixed $value
      */
-    public function setParam($id, $value)
-    {
-        $this->parameters[$id] = $value;
-    }
+    public function setParam($id, $value);
 
     /**
      * Gets a param from the container or a default value.
      *
      * @param string $id
-     * @param mixed  $default
+     * @param mixed $default
      *
      * @return mixed
      */
-    public function getParam($id, $default = null)
-    {
-        return isset($this->parameters[$id]) ? $this->parameters[$id] : $default;
-    }
+    public function getParam($id, $default = null);
 
     /**
-     * Sets a object or a callable for the object creation. A callable will be invoked
-     * every time get is called.
+     * Sets a object to be used as a service
      *
-     * @param string          $id
-     * @param object|callable $value
+     * @param string $id
+     * @param object $value
      *
      * @throws \InvalidArgumentException if service is not an object or callable
```

Comment:
it should not allow callables anymore, as definitions are handled by a separate method